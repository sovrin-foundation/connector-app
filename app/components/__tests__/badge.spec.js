// @flow
import React from 'react'
import 'react-native'
import Badge from '../badge'
import renderer from 'react-test-renderer'

describe('<Badge />', () => {
  function getProps() {
    return {
      count: 10,
      secondary: false,
      badgeStyle: {},
      shadow: false,
      onPress: jest.fn(),
    }
  }

  it('should render light colored badge', () => {
    const props = getProps()
    const badge = renderer.create(<Badge {...props} center={true} />).toJSON()
    expect(badge).toMatchSnapshot()
  })

  it('should render dark colored badge', () => {
    const props = getProps()
    const badge = renderer.create(<Badge {...props} text={10} />).toJSON()
    expect(badge).toMatchSnapshot()
  })
})
