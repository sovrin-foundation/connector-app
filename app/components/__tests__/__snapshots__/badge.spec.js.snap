// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render dark colored badge 1`] = `
<View
  clearBg={true}
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <Image
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onPress={[MockFunction]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 26,
        "width": 22,
      }
    }
    testID={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    center={true}
    ellipsizeMode="tail"
    secondary={false}
    style={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "bold",
        },
        Object {
          "color": "#f0f0f0",
        },
        Object {
          "textAlign": "center",
          "textAlignVertical": "center",
        },
        Object {},
      ]
    }
    text={10}
  >
    10
  </Text>
</View>
`;

exports[`<Badge /> should render light colored badge 1`] = `
<View
  clearBg={true}
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <Image
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onPress={[MockFunction]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 26,
        "width": 22,
      }
    }
    testID={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    center={true}
    ellipsizeMode="tail"
    secondary={false}
    style={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "bold",
        },
        Object {
          "color": "#f0f0f0",
        },
        Object {
          "textAlign": "center",
          "textAlignVertical": "center",
        },
        Object {},
      ]
    }
    text={10}
  >
    10
  </Text>
</View>
`;
